action: use official github action for golangci-lint

Desiki-high opened the pull request:
## Relevant Issue (if applicable)
N/A.

## Details
1. separate contrib-test and golang lint by `contrib-lint`.
2. add contrib-lint in smoke test. So we don't need to install `golangci-lint` manually.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

adam3q reviewed (commented):
Thanks for your contribution! Can you squash these commits first?

imeoer reviewed (approved):
Thanks, LGTM!

Desiki-high committed:
action: add contrib-lint in smoke test

1. Use the official GitHub action for golangci-lint from its authors.
2. fix golang lint error with v1.56
3. separate test and golang lint.Sometimes we need tests without golang lint and sometimes we just want to do golang lint.

Signed-off-by: Yadong Ding <ding_yadong@foxmail.com>
